lib/plugins/aws/package/compile/events/apiGateway/lib/validate.js

```diff
@@ -213,6 +213,12 @@ module.exports = {
     if (typeof authorizer === 'string') {
       if (authorizer.toUpperCase() === 'AWS_IAM') {
         type = 'AWS_IAM';
+      } else if (authorizer.toUpperCase() === 'API_KEY') {
+        // this is an alias for private: true in the event. If API_KEY is
+        // defined, we return nothing and set the event to being private, so it
+        // will require a valid api key further in the chain
+        http.private = true;  // eslint-disable-line no-param-reassign
+        return null;
```

Comment:
Why `return null`?
I think this is unnecessary. Am I wrong?

Reply:
Well, not entirely. 

This is part of the `getAuthorizer` call, which is called here: https://github.com/svdgraaf/serverless/blob/6e38975c4150723d479213ae40bfc529a63cc21d/lib/plugins/aws/package/compile/events/apiGateway/lib/validate.js#L50

Which sets the value of `http.authorizer` to whatever the `getAuthorizer` returns. Whenever that is set to something. It's going to trigger in some other places, like:

https://github.com/svdgraaf/serverless/blob/6e38975c4150723d479213ae40bfc529a63cc21d/lib/plugins/aws/package/compile/events/apiGateway/lib/method/authorization.js#L15

So I need to return something to exit the function early. If I return an empty hash, the `http.authorizer` is still set, which will trigger at some more points.

I could also move the check for `api_key` one layer up to here: https://github.com/svdgraaf/serverless/blob/6e38975c4150723d479213ae40bfc529a63cc21d/lib/plugins/aws/package/compile/events/apiGateway/lib/validate.js#L49

But looks a bit silly, all on it's own, and seems out of place.

Is there perhaps a better way to not unset the `http.authorizer` there, instead of returning null? My node skills are not so good :)